### PR TITLE
replace ANSIBLE_ like envs with their value when generating ansible.cfg

### PR DESCRIPTION
**Pull request by ingvagabund**
ANSIBLE_ like environment variables have higher precedence over their equivalents inside ansible.cfg. Thus, if the envs are set on CLI, the roles_path, library and filter_plugins inside ansible.cfg are completely ignored.
    
Given the ansible.cfg is generated each time `molecule test` command is run, it is ok to replace the envs with they values and filtering all three ANSIBLE_ like envs when generating ansible-playbook CLI command.

Fixes: #845

### CODE REVIEW

**Comment by ingvagabund** [comment]
@retr0h

File: molecule/provisioner/ansible.py [hunk 1]
```diff
@@ -206,18 +206,40 @@ def default_config_options(self):
 
         :return: dict
         """
+        envs = self._config.config['provisioner']['env']
+
+        roles_path = '../../../../'
```
Comment (ashcrow) [inline molecule/provisioner/ansible.py]: Will this always be accurate? (4 dirs back)

**Comment by retr0h** [comment]
I highly suggest looking at molecule v2 branch.  This is the future of the project as v1 is on the way out.

**Comment by retr0h** [comment]
Think we will need some unit tests reproducing these overriden scenarios.

**Comment by retr0h** [comment]
Implemented at 38e8acb.

**Pull request closed without merging**